test/test-conveniences.c

```diff
@@ -29,4 +41,1054 @@ bson_iter_bson (bson_iter_t *iter, bson_t *bson)
    }
    BSON_ASSERT (data);
    bson_init_static (bson, data, len);
+}
```

Comment:
Feel free to ignore changes in this file. This is entirely copied from libmongoc's test-conveniences.c to copy in a recursive bson matching assertion. This will likely be useful for future testing (MONGOCRYPT-275 in particular).

Reply:
Thanks